jobsdb/unionQuery.go

```diff
@@ -89,37 +88,35 @@ func (mj *MultiTenantHandleT) GetAllJobs(ctx context.Context, workspaceCount map
 		StateFilters:                  params.StateFilters,
 	}
 	start := time.Now()
+	dsQueryCount := 0
+	dsQueryLimit := mj.dsLimit
 	for _, ds := range dsList {
-		jobs, err := mj.getUnionDS(ctx, ds, workspaceCount, workspacePayloadLimitMap, conditions)
+		if (dsQueryLimit > 0 && dsQueryCount >= dsQueryLimit) || len(workspaceCount) == 0 {
+			break
+		}
+		jobs, dsHit, err := mj.getUnionDS(ctx, ds, workspaceCount, workspacePayloadLimitMap, conditions)
 		if err != nil {
 			return nil, err
 		}
 		outJobs = append(outJobs, jobs...)
-		if len(jobs) > 0 {
-			tablesQueried++
-		}
-
-		if len(workspaceCount) == 0 {
-			break
-		}
-		if tablesQueried >= maxDSQuerySize {
-			break
```

Comment:
Propose to leave the old logic intact. The intention is not to change unionQuery's behaviour

Reply:
maxDSQuerySize is the same as DSLimit. So I removed it. Should we support both of them?

Reply:
according to @cisse21 response [here](https://github.com/rudderlabs/rudder-server/pull/2446#discussion_r974485327), it is still desirable to use a default limit of 10 for unionQuery, thus we shouldn't consider the two limits (`maxDSQuerySize` & `dsLimit`) as same.

Reply:
In that case, we should keep the default limit of dsLimit as 10 for unionQuery right? Or we should get rid of `dsLimit` altogether

Reply:
let's treat them as different things, see [here](https://github.com/rudderlabs/rudder-server/pull/2446#discussion_r974068382)

Reply:
@cisse21 they may look the same but they are practically not, since we don't want to enforce a default limit of 10 datasets for queries other than unionDS.
IMO by default there should be no limit to the number of datasets that we can query as this limit might hinder proper job pickup and processing.
I see this limit only as a tool to use when things are going wrong and there is a lot of pressure to the database, where we can temporarily impose such a limit, if necessary, but still we need to revert it after the incident is resolved.